tests/PrefixCacheBootstrapperTest.php

```diff
@@ -0,0 +1,382 @@
+<?php
+
+declare(strict_types=1);
+
+use Illuminate\Cache\CacheManager;
+use Illuminate\Support\Facades\Event;
+use Stancl\Tenancy\Events\TenancyEnded;
+use Stancl\Tenancy\Tests\Etc\CacheService;
+use Stancl\Tenancy\Events\TenancyInitialized;
+use Stancl\Tenancy\Listeners\BootstrapTenancy;
+use Stancl\Tenancy\Listeners\RevertToCentralContext;
+use Stancl\Tenancy\Tests\Etc\SpecificCacheStoreService;
+use Stancl\Tenancy\Bootstrappers\PrefixCacheTenancyBootstrapper;
+
+beforeEach(function () {
+    config([
+        'tenancy.bootstrappers' => [
+            PrefixCacheTenancyBootstrapper::class
+        ],
+        'cache.default' => $cacheDriver = 'redis',
+        'cache.stores.redis2' => config('cache.stores.redis'),
+    ]);
+
+    PrefixCacheTenancyBootstrapper::$tenantCacheStores = [$cacheDriver];
+    PrefixCacheTenancyBootstrapper::$prefixGenerator = null;
+
+    config('tenancy.cache.manager')::$addTags = false;
```

Comment:
This line doesn't really make sense to me. As I wrote in the other review, I'd get rid of this approach to the config completely, but even if we did specify the cache manager in the config, using it like this seems wrong. **Our class** has the `addTags` property. Dynamically fetching the class from the config doesn't seem to add anything here, it just hides the `CacheManager` class (which will always be used here I think?) behind an unnecessary abstraction layer.

Reply:
And in the `Tenancy overrides CacheManager` test, just reference the two classes directly. Laravel's CacheManager and our CacheManager (aliased to `TenantCacheManager` or something like that).

Reply:
I changed the approach so that the config isn't used, and we override CacheManager only when using CacheTagsBootstrapper

https://github.com/archtechx/tenancy/pull/1014#discussion_r1170915056